### ISSUE TITLE
chmod doesn't work on Windows

### ISSUE THREAD

**Issue by cknight** [denoland/deno#4357]
In Node, the following is observed on Windows:

```typescript
const fs = require('fs');
fs.chmodSync('test_file.txt', 0o000);  //makes the file read-only
fs.chmodSync('test_file.txt', 0o777);  //makes the file writable
```
However, in Deno no file mode I tested changed the read-only attribute in any way (including the above two file modes).

From the [Node docs](https://nodejs.org/api/fs.html#fs_fs_chmod_path_mode_callback):
> on Windows only the write permission can be changed, and the distinction among the permissions of group, owner or others is not implemented.

Windows is not mentioned in the [Deno docs](https://deno.land/typedoc/index.html#chmod).

It would be good if Deno could manipulate the read-only attribute of Windows files and specifically call out Windows quirks in the documentation for chmod.

**Comment by dubiousjim** [comment]
Sounds reasonable.

Should presumably also apply to the soon-to-be added `mode` option on `open` (#4289, should be merged soon) and `truncate` (#4288, waiting on some other PRs). Perhaps also the `mode` option on `mkdir`, but I've got little Windows experience so I don't know if they accept/honor the read-only attribute on directories.

Should the API be that `0o000` and `0o777` are required to change these attributes, and other values are ignored? Or that other values throw an error? Or that only the user-write bit (`0o200`) has an effect, and other bits are ignored?

**Comment by dubiousjim** [comment]
Of course, calling `truncate` with a read-only `mode` is guaranteed to either be ignored or fail. Still if the `mode` option is interpreted a given way on Windows for `open`, it should be interpreted that way for `truncate` too.

**Comment by cknight** [comment]
> Should the API be that `0o000` and `0o777` are required to change these attributes, and other values are ignored? Or that other values throw an error? Or that only the user-write bit (`0o200`) has an effect, and other bits are ignored?

I think the way node handles this is that only the user write-bit is honored.  In my example `0o000` and `0o777` were arbitrary.  `0o077` and `0o007` both behave as per `0o000` for example.  Caveat, my access to a windows test box is limited, so I'm going off memory here.

**Comment by TradeIdeasPhilip** [comment]
I see similar issues in Deno.writeTextFile().  I'm setting the mode to 0o444 (read only).  The file is created and filled correctly.  But the permissions are wrong.  They are "-rw-r--r--" (644) according to git bash, and all permissions for the current use according to Windows file explorer.

I should have created a read only file.  Barring that I should have gotten an exception or at least seen more details in the documentation.
